unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/Settings/PlayerPrefsQualitySettingsRepository.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using UnityEngine;
+
+namespace DCL.SettingsCommon
+{
+    public class PlayerPrefsQualitySettingsRepository : ISettingsRepository<QualitySettings>
+    {
+        private readonly IPlayerPrefsSettingsByKey settingsByKey;
+        private readonly QualitySettings defaultSettings;
+        private QualitySettings currentSettings;
+        
+        public event Action<QualitySettings> OnChanged;
+
+        public PlayerPrefsQualitySettingsRepository(
+            IPlayerPrefsSettingsByKey settingsByKey,
+            QualitySettings defaultSettings)
+        {
+            this.settingsByKey = settingsByKey;
+            this.defaultSettings = defaultSettings;
+            currentSettings = Load();
+        }
+
+        public QualitySettings Data => currentSettings;
+
+        public void Apply(QualitySettings settings)
+        {
+            if (currentSettings.Equals(settings)) return;
+            currentSettings = settings;
+            OnChanged?.Invoke(currentSettings);
+        }
+
+        public void Reset()
+        {
+            Apply(defaultSettings);
+        }
+
+        public void Save()
+        {
+            settingsByKey.SetString("displayName", currentSettings.displayName);
```

Comment:
I'm seeing that the setting names are re-used on a few places, I'd put those into constants to avoid misspelling issues.

Reply:
done!